template_handler/cdk.py

```diff
@@ -0,0 +1,218 @@
+            AWS_SECRET_ACCESS_KEY=credentials.secret_key,
+        )
+
+        # There might not be a session token, so if there isn't one, make sure it doesn't exist in
+        # the envs being passed to the subprocess
+        if credentials.token is None:
+            envs.pop('AWS_SESSION_TOKEN', None)
+        else:
+            envs['AWS_SESSION_TOKEN'] = credentials.token
+
+        return envs
+
+    def _cmd_exists(self, cmd) -> bool:
+        """
+        Checks whether a specified CLI command exists
+        """
+
+        cmd_exists = shutil.which(cmd) is not None
+        self.logger.debug(f"command '{cmd}' exists: {cmd_exists}")
+        return cmd_exists
+
+    def _node_package_exists(self, package) -> bool:
+        """
+        Checks whether a specifie Node package exists either in the workspace or global scope
+        """
+
+        package_exists = False
+        workspace_result = subprocess.run(
+            f'npm list {package}',
+            shell=True,
+            capture_output=True
+        )
+        self.logger.debug(f"Workspace NPM package '{package}' exists: {not bool(workspace_result.returncode)}")
+
+        if workspace_result.returncode == 0:
+            package_exists = True
+        else:
+            global_result = subprocess.run(
+                f'npm --global list {package}',
+                shell=True,
+                capture_output=True
+            )
+            self.logger.debug(f"f'Global NPM package '{package}' exists: {not bool(global_result.returncode)}")
+
+            if global_result.returncode == 0:
+                package_exists = True
+
+        return package_exists
+
+    def _check_prerequisites(self) -> None:
+        """
+        Checks the command and Node package requirements for the handler
+        """
+
+        # Check Command Prerequisites
+        cmd_prerequisites = [
+            'node',
+            'npx'
+        ]
+        for cmd_prerequisite in cmd_prerequisites:
+            if not self._cmd_exists(cmd_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Command prerequisite '{cmd_prerequisite}' not found")
+
+        # Check Node Package Prerequisites
+        node_prerequisites = [
+            'cdk-assets'
+        ]
+        for node_prerequisite in node_prerequisites:
+            if not self._node_package_exists(node_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Node Package prerequisite '{node_prerequisite}' not found")
+
+    def handle(self) -> str:
+        """
+        Main Sceptre CDK Handler function
+
+        Returns
+        -------
+        str|bytes
+            CDK synthesised CloudFormation template
+        """
+
+        self.logger = self.PrefixLoggerAdapter(self.logger, {'prefix': self.name})
+
+        self._check_prerequisites()
+
+        # Import CDK Python template module
+        template_path = str(pathlib.Path('templates', self.cdk_template_path))
```

Comment:
Fun tidbit about pathlib.Path... they've overloaded the division operator which makes it super handy to do this
```suggestion
        template_path = str(pathlib.Path('templates') / self.cdk_template_path)
```

Not saying you need to do that. I just fine it a very useful way to construct paths.